### ISSUE TITLE
Multiple class detection on the same area

### ISSUE THREAD

**Issue by ngito** [AlexeyAB/darknet#2061]
Hi all,

I'm in the middle of creating custom datasets to identify aeroplane & airline name based on its logo.
I have labelled about 70 images for each airlines - total 210 images for 3 airlines, 50 images for training, 20 for validation - I know this is very small datasets, planning to add to 200 images / class

It has been working as expected, however when I added the fourth class, aeroplane class & datasets from VOC (I included 1121 images), darknet could not detect aeroplane class and airline at the same time.

However, If I test individual class (aeroplane, and airline), both detection are working perfectly, but it can not detect both class at the same time, example: detect an object is aeroplane and detect its airline name based on the logo on aeroplane's body.

Is it because aeroplane area is overlapped with airline logo causing this problem?

About my training parameters:
- cfg/yolov3 cfg file as baseline
- darknet53.conv.74 as pre train weights
- I stopped training when avg loss = 0.098631, iteration 6900+

Many thanks for your feedback

**Comment by AlexeyAB** [comment]
@ngito Hi,

Can you show examples of images with detected objects?

**Comment by ngito** [comment]
Hi Alexey,

Thanks for replying...

All these images were inferenced using my custom weight, the last aeroplane image is taken from VOC.

![a-score](https://user-images.githubusercontent.com/17381377/50052277-16a05e00-0154-11e9-8ff6-2a460393c5d6.jpg)
![g-score](https://user-images.githubusercontent.com/17381377/50052278-1738f480-0154-11e9-8160-3a1deef6f699.jpg)
![m-score](https://user-images.githubusercontent.com/17381377/50052279-1738f480-0154-11e9-93a0-c70ad4f9405a.jpg)
![aeroplane-score](https://user-images.githubusercontent.com/17381377/50052298-63843480-0154-11e9-8741-06b39ac7ce8f.jpg)

**Comment by AlexeyAB** [comment]
Do you want such detection as on image that I have attached?

So you should label `object: aeroplane` on each Training image where is aeroplane: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> **check that each object are mandatory labeled in your dataset - no one object in your data set should not be without label.** In the most training issues - there are wrong labels in your dataset (got labels by using some conversion script, marked with a third-party tool, ...). Always check your dataset by using: https://github.com/AlexeyAB/Yolo_mark

![50052298-63843480-0154-11e9-8741-06b39ac7ce8f](https://user-images.githubusercontent.com/4096485/50052357-d7faab80-0133-11e9-858b-ccf22e5aff37.jpg)

**Comment by AlexeyAB** [comment]
@ngito So you should label it in the same way:

Label it:
![50052278-1738f480-0154-11e9-8160-3a1deef6f699](https://user-images.githubusercontent.com/4096485/50052553-f1512700-0136-11e9-97ae-50aa5586543e.jpg)


Instead of:
![50052278-1738f480-0154-11e9-8160-3a1deef6f699 1](https://user-images.githubusercontent.com/4096485/50052562-0332ca00-0137-11e9-9a2e-50cf4397be37.jpg)

**Comment by ngito** [comment]
@AlexeyAB 
Thanks for your quick reply, I will retrain my weights again, I will let you know the result.

**Comment by ngito** [comment]
@AlexeyAB 

It works as expected now, many thanks!

Just few more questions:
1. I noticed, i got reasonable good results after avg loss around 0.1x, while I found continue training until avg loss around 0.0x took much longer time, while I'm not quite sure the significant benefit if continue to get the lowest avg loss. Based on your experience, is it worthed to continue training until reaching lowest avg loss? Should I compare the mAP based on different `.weights` file?

2. Just wondering how long did you train entire VOC datasets? What gear do you use?

Many Thanks!

**Comment by AlexeyAB** [comment]
@ngito 

1. Better to train with flag `-map`
`./darknet detector train data/obj.data yolo-obj.cfg darknet53.conv.74 -map`

So you will see mAP value on `valid.txt` during training on `train.txt` (txt-files that are specified `obj.data` file)

When accuracy stops increasing, then stop training.

![68747470733a2f2f6873746f2e6f72672f776562742f79642f766c2f61672f7964766c616775746f66327a636e6a6f64737467726f656e3861632e6a706567](https://user-images.githubusercontent.com/4096485/50085275-9db00d80-020a-11e9-970b-646595bda39b.jpg)


2. I use Quadro P5000. For different datasets and models it takes several days.

**Comment by ngito** [comment]
@AlexeyAB many thanks for your response and your time!

**Comment by ngito** [comment]
@AlexeyAB 
Sorry 1 last question :D Can you tell me how to generate that chart when I resume training from previously cancelled job? Thanks
